### ISSUE TITLE
about the loss of the global descriptor

### ISSUE THREAD

**Issue by ghoshaw** [shamangary/awesome-local-global-descriptor#9]
Hi, Thanks for the sharing.
     In your experiments,  you concat the tensor from FPN and do L2-norm then calc the contrastloss, or only use L2-norm in test phase?

**Comment by shamangary** [comment]
Hello. For local descriptor, L2-norm was adopted in the testing stage. In the training stage we use the way d2-net use, basically using softmax as a soft-detection with NMS for the training.

For global descriptor, it is like a second part of the network which has nothing to do with the L2-norm. The descriptor is the concatenation of the FPN outputs, and we use contrastive loss to train it.

**Comment by ghoshaw** [comment]
Hi, thanks for your answer, but my question is about the global descriptor, 
![image](https://user-images.githubusercontent.com/23494873/76415361-05d0ad80-63d4-11ea-9b15-91bad2baaeb7.png)

**Comment by ghoshaw** [comment]
In paper "Fine-tuning CNN Image Retrieval with No Human Annotation", it said 
![image](https://user-images.githubusercontent.com/23494873/76415504-492b1c00-63d4-11ea-9511-bb81e0b44b0b.png), but in your paper, it seems to calc the contrastloss use the GEM result directly.

**Comment by shamangary** [comment]
Hello @ghoshaw, the global descriptor is L2-normalized. What I meant was L2-norm (a single value) was adopted in our local keypoint detection. But yes, every feature should be normalized.

**Comment by ghoshaw** [comment]
OK，thanks very much!
